### ISSUE TITLE
/usr/lib64/python2.7/site-packages/cffi/model.py:526: UserWarning: 'git_checkout_notify_t' has no values explicitly defined;

### ISSUE THREAD

**Issue by tjyang** [saltstack/salt#38275]
### Description of Issue/Question
Seeing warning message in this issue's title

### Steps to Reproduce Issue
restart or do a versions-report on salt-minion/salt-master will generate this message

### Versions Report

```
[root@salt01t ~]# salt-master --versions-report
/usr/lib64/python2.7/site-packages/cffi/model.py:526: UserWarning: 'git_checkout_notify_t' has no values explicitly defined; next version will refuse to guess which integer type it is meant to be (unsigned/signed, int/long)
  % self._get_c_name())
/usr/lib64/python2.7/site-packages/cffi/model.py:526: UserWarning: 'git_merge_tree_flag_t' has no values explicitly defined; next version will refuse to guess which integer type it is meant to be (unsigned/signed, int/long)
  % self._get_c_name())
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: 1.6.0
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.21.0
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.21.4
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-514.2.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

[root@salt01t ~]#

```

**Comment by Ch3LL** [comment]
@tjyang are you using pygit2 as well?

I think this is a duplicate of #29049 and is not an issue with salt but that warning is coming from cffi and pygit2. As stated in that issue newer pygit2 issues might not have this issue.

**Comment by ucsdphysics** [comment]
The reason for this is the same as what happened in https://github.com/saltstack/salt/issues/29049, and comes from doing an upgrade of CentOS 7.2 to 7.3. The version of python-cffi got a version bump from 0.8.6 to 1.6.0, while python-pygit2 stayed at 0.21.4 and libgit2 stayed at 0.21.5. The newer cffi version causes the warning to be generated with the older version of pygit2.

I was able to resolve this by building new packages for both python-pygit2 (0.24.2) and libgit2 (0.24.3) using SRPMS from Fedora 25 (with some minor modifications to work in CentOS) and installing them.

**Comment by Ch3LL** [comment]
@ucsdphysics thanks for the feedback and workaround. @tjyang can you try updating the packages? Closing for now since this is confirmed by other user as well to not be an issue with salt. But also want to make sure workaround is working @tjyang so please ping me on this issue if it does not.

**Comment by tjyang** [comment]
@Ch3LL, I didn't update my python-cffi package from Fredora25.
Looks like in 2016.11.2, the warning message was silenced. 
Anyway, thanks for your dev team's work.

```
Salt Version:
           Salt: 2016.11.2

Dependency Versions:
           cffi: 0.8.6
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: 1.3.8
         Jinja2: 2.8
        libgit2: 0.21.0
        libnacl: 1.4.3
       M2Crypto: 0.21.1
           Mako: 0.8.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.3
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.21.4
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.36.3.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core

[me@salt01 ~]$ rpm -qa |grep python-cffi
python-cffi-0.8.6-2.el7.x86_64
[me@salt01 ~]$

```

**Comment by tjyang** [comment]
Sorry, I was not paying too much attention on this subject. my previous post has no warning message was because I didn't upgrade centos 7.2 to 7.3.

Although this is a  upstream(centos 7.3) bug. Is there anyway from salt's side to silent these two  warning messages ?

**Comment by Ch3LL** [comment]
ping @terminalmage do you know if there is anything we can do on our end to get rid of this warning as a workaround even though this is an upstream issue?

**Comment by terminalmage** [comment]
@Ch3LL See https://github.com/saltstack/salt/pull/39871

**Comment by tjyang** [comment]
@terminalmage and @Ch3LL , thanks for the understanding and providing  this fix.

**Comment by tjyang** [comment]
Can #39871 be included into 2016.11.4 ?

**Comment by terminalmage** [comment]
@tjyang Yes, it will automatically be included in 2016.11.4.
